DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -812,85 +819,44 @@ function Update-WebsiteBinding
     }
 }
 
-function Get-WebBindingObject
-{
-    param
-    (
-        $BindingInfo
-    )
-
-    #First split properties by ']:'. This will get IPv6 address split from port and host name
-    $Split = $BindingInfo.BindingInformation.split('[]')
-    if($Split.count -gt 1)
-    {
-        $IPAddress = $Split.item(1)
-        $Port = $Split.item(2).split(':').item(1)
-        $HostName = $Split.item(2).split(':').item(2)
-    }
-    else
-    {
-        $SplitProps = $BindingInfo.BindingInformation.split(':')
-        $IPAddress = $SplitProps.item(0)
-        $Port = $SplitProps.item(1)
-        $HostName = $SplitProps.item(2)
-    }
-
-    return New-Object -TypeName PSObject -Property @{
-        Protocol              = $BindingInfo.protocol
-        IPAddress             = $IPAddress
-        Port                  = $Port
-        HostName              = $HostName
-        CertificateThumbprint = $BindingInfo.CertificateHash
-        CertificateStoreName  = $BindingInfo.CertificateStoreName
-        sslFlags              = $BindingInfo.sslFlags
-    }
-}
 
 # Helper function used to Update default pages of website
 function Update-DefaultPages
 {
     param
-   (
-        [string] $Name,
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $Name,
 
-        [string[]] $DefaultPage
+        [Parameter(Mandatory = $true)]
+        [String[]]
+        $DefaultPage
     )
 
-    $allDefaultPage = @(Get-WebConfiguration //defaultDocument/files/* -PSPath (Join-Path -Path 'IIS:\sites\' -ChildPath $Name) | ForEach-Object -Process {
-        Write-Output -InputObject $_.value
-    })
+    $allDefaultPage = @(
+        Get-WebConfiguration -Filter '//defaultDocument/files/*' -PSPath "IIS:\Sites\$Name" |
+        ForEach-Object -Process {
+            Write-Output -InputObject $_.value
+        }
+    )
 
-    foreach($page in $DefaultPage)
+    foreach ($Page in $DefaultPage)
     {
-        if(-not ($allDefaultPage -icontains $page))
+        if ($allDefaultPage -inotcontains $Page)
         {
-            Write-Verbose -Message ("Deafult page for website $Name has been updated to $page")
-            Add-WebConfiguration //defaultDocument/files -PSPath (Join-Path -Path 'IIS:\sites\' -ChildPath $Name) -Value @{
-                value = $page
+            Add-WebConfiguration -Filter '//defaultDocument/files' -PSPath "IIS:\Sites\$Name" -Value @{
+                value = $Page
+            }
+
+            if ($? -eq $true)
+            {
+                Write-Verbose -Message "Default page for website $Name has been updated to $Page"
             }
         }
     }
 }
 
-function Write-Log
```

Comment:
I'd like to not get rid of this function, and would honestly prefer the verbose stream uses it, it's been very handy tracking down some of these issues as the verbose stream occasionally sort circuts.